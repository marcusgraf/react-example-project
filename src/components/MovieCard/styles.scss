@import '../../theme/variables';

.MovieCard {
  position: relative;
  background-repeat: no-repeat;
  background-size: cover;
  width: 100%;
  padding: 10em 0;

  &::before {
    background-color: rgba(0, 0, 0, 0.4);
    content: '';
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
  }

  .banner_content_wrapper {
    width: 900px;
    max-width: 90%;
    margin: 0 auto;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .poster {
    width: 275px;
    position: relative;
    float: left;

    .featured_image {
      width: 100%;
      max-width: 100%;
      border-radius: 0.5em;
      -webkit-box-shadow: 0 0 76px 0 rgba(0, 0, 0, 0.47);
      -moz-box-shadow: 0 0 76px 0 rgba(0, 0, 0, 0.47);
      box-shadow: 0 0 76px 0 rgba(0, 0, 0, 0.47);
    }

    .play_button {
      position: absolute;
      width: 80px;
      left: 50%;
      top: 50%;
      margin: -40px 0 0 -40px;
    }
  }

  .content {
    float: left;
    max-width: 500px;
    margin-left: 10%;
    position: relative;

    .title {
      display: inline;
      font-size: 1.75em;
      color: $white;
    }

    .description {
      color: #bbbebf;
      font-size: 1em;
      line-height: 2;
    }

    .info {
      color: $white;
      font-size: 0.8em;
      font-weight: 700;
      margin-top: 3em;

      span {
        margin: 0.5em;
      }
    }
  }
}
