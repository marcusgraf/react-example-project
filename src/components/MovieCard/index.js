/* @flow */

import React from 'react';
import type { Element } from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scss';

type Props = { movie: PropTypes.object };

const MovieCard = ({ movie }: Props): Element<'div'> => (
  <div className={styles.MovieCard} style={{ backgroundImage: `url("${movie.full_backdrop_path}")` }}>
    <div className={styles.banner_content_wrapper}>
      <div className={styles.poster}>
        <img src={movie.full_poster_path} alt={movie.title} className={styles.featured_image} />
        <img src="https://res.cloudinary.com/dw369yzsh/image/upload/v1470916845/play_button_ngnw1z.png" alt="Play Trailer" className={styles.play_button} />
      </div>
      <div className={styles.content}>
        <h2 className={styles.title}>{movie.title}</h2>

        <p className={styles.description}>
          {movie.overview}
        </p>

        <p className={styles.info}>
          {movie.vote_count} Views<span>|</span>
          {movie.release_date}<span>|</span>
          {movie.vote_average} / 10 Points
        </p>
      </div>
    </div>
  </div>
);

export default MovieCard;
