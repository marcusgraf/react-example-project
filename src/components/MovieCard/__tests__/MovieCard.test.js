import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';

import MovieCard from '../index';

describe('<MovieCard />', () => {
  test('renders', () => {
    const mockData = JSON.parse('{"vote_count":7141,"id":11,"video":false,"vote_average":8.1,"title":"Star Wars",' +
      '"popularity":29.242203,"poster_path":"/btTdmkgIvOi0FFip1sPuZI2oQG6.jpg","original_language":"en",' +
      '"original_title":"Star Wars","genre_ids":[12,28,878],"backdrop_path":"/c4zJK1mowcps3wvdrm31knxhur2.jpg",' +
      '"adult":false,"overview":"Princess Leia is captured and held hostage by the evil Imperial forces in ' +
      'their effort to take over the galactic Empire. Venturesome Luke Skywalker and dashing captain Han ' +
      'Solo team together with the loveable robot duo R2-D2 and C-3PO to rescue the beautiful princess and ' +
      'restore peace and justice in the Empire.","release_date":"1977-05-25","full_poster_path":' +
      '"https://image.tmdb.org/t/p/w500//btTdmkgIvOi0FFip1sPuZI2oQG6.jpg","full_backdrop_path":"' +
      'https://image.tmdb.org/t/p/w500//c4zJK1mowcps3wvdrm31knxhur2.jpg","' +
      'listening_token":"05159528-6c9a-45f1-a840-3167340d751d"}');
    const tree = renderer.create(
      <StaticRouter context={{}}>
        <MovieCard movie={mockData} />
      </StaticRouter>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
