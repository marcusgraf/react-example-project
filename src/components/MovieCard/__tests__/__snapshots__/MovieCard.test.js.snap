// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCard /> renders 1`] = `
<div
  className="MovieCard"
  style={
    Object {
      "backgroundImage": "url(\\"https://image.tmdb.org/t/p/w500//c4zJK1mowcps3wvdrm31knxhur2.jpg\\")",
    }
  }
>
  <div
    className="banner_content_wrapper"
  >
    <div
      className="poster"
    >
      <img
        alt="Star Wars"
        className="featured_image"
        src="https://image.tmdb.org/t/p/w500//btTdmkgIvOi0FFip1sPuZI2oQG6.jpg"
      />
      <img
        alt="Play Trailer"
        className="play_button"
        src="https://res.cloudinary.com/dw369yzsh/image/upload/v1470916845/play_button_ngnw1z.png"
      />
    </div>
    <div
      className="content"
    >
      <h2
        className="title"
      >
        Star Wars
      </h2>
      <p
        className="description"
      >
        Princess Leia is captured and held hostage by the evil Imperial forces in their effort to take over the galactic Empire. Venturesome Luke Skywalker and dashing captain Han Solo team together with the loveable robot duo R2-D2 and C-3PO to rescue the beautiful princess and restore peace and justice in the Empire.
      </p>
      <p
        className="info"
      >
        7141
         Views
        <span>
          |
        </span>
        1977-05-25
        <span>
          |
        </span>
        8.1
         / 10 Points
      </p>
    </div>
  </div>
</div>
`;
